c++/src/kj/compat/http.c++

```diff
@@ -2637,6 +2637,31 @@ public:
 
   uint64_t receivedByteCount() override { return receivedBytes; }
 
+  kj::Maybe<kj::String> getPreferredExtensions(ExtensionsContext ctx) override {
+    if ((maskKeyGenerator == nullptr && ctx == ExtensionsContext::REQUEST) ||
+        (maskKeyGenerator != nullptr && ctx == ExtensionsContext::RESPONSE)) {
+      // We can only direct pump from a client to a server (and vice versa).
+      // If the caller attempts to get preferred extensions for pumping from client to client,
+      // or from server to server, then we will return nullptr, indicating no preferred extensions.
+      return nullptr;
+    }
+    KJ_IF_MAYBE(config, compressionConfig) {
+      if (ctx == ExtensionsContext::RESPONSE) {
+        // We are a server websocket, so our
```

Comment:
This comment is a little confusing. The caller is asking to construct the headers for a new server-side websocket, buht that doesn't mean "we" are a server websocket -- in fact, "we" must be a client websocket for this to work?